### PR TITLE
FEAT: add starter workflow for Stale PRs

### PR DESCRIPTION
**Pull request by markrmullan**
## Context

> Why are you making this change?

- Automate the process of closing abandoned PRs in any repo that would like to opt in to this functionality.

## Origin

> What team/individual requested this change? Please link to a JIRA ticket or other documentation if applicable. If you identified this issue yourself please write "Me" (and still link to any additional context if it exists).

Me

## Summary of Changes

This PR adds a starter workflow that other repositories can opt into. We'll enable this for a repo, verify it's working as intended, and can then roll it out to others.

## Test Plan

- Enable this workflow in `code` or `web`
- Take a look at some old PRs in `code` or `web` tomorrow morning, see if they are properly labeled + commented on

## Documentation

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to external documentation if necessary (READMEs, Confluence, etc.)

### CODE REVIEW

File: workflow-templates/stale_prs.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/stale@v4.1.0
+        # https://github.com/actions/stale#all-options
+        with:
+          stale-pr-message: '⚠️ This PR has been labeled as Stale after 60 days of inactivity, and will be automatically closed in 30 days. Questions / concerns? Please reach out to your friendly neighborhood #developer-experience team.'
+          # PRs with no activity for 60 days will be labeled as Stale
+          days-before-pr-stale: 60
+          # PRs that are Stale for 30 days will be closed
+          days-before-pr-close: 30
```
Comment (markrmullan) [inline workflow-templates/stale_prs.yaml:30]: These calls are authenticated using a GH token that has a lifetime of the duration of this job being executed https://docs.github.com/en/actions/security-guides/automatic-token-authentication. In other words, extracting this functionality out of the lgtm bot will be a net positive w.r.t. GH rate limiting.

File: workflow-templates/stale_prs.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    - cron: '20 9 * * *'
+
+jobs:
+  stale_prs:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/stale@v4.1.0
+        # https://github.com/actions/stale#all-options
+        with:
+          stale-pr-message: '⚠️ This PR has been labeled as Stale after 60 days of inactivity, and will be automatically closed in 30 days. Questions / concerns? Please reach out to your friendly neighborhood #developer-experience team.'
```
Comment (nathanshelly) [inline workflow-templates/stale_prs.yaml]: assuming this is markdown maybe link to the channel?

```suggestion
          stale-pr-message: '⚠️ This PR has been labeled as Stale after 60 days of inactivity, and will be automatically closed in 30 days. Questions / concerns? Please reach out to your friendly neighborhood [#developer-experience](https://opendoor.slack.com/archives/C6PGFGTV0) team.'
```

**Review by nathanshelly — APPROVED** [review]
looks good to go ahead and test to me!

File: workflow-templates/stale_prs.yaml (hunk 2 above)
Comment (markrmullan) [reply workflow-templates/stale_prs.yaml]: nice great call! 💯

**Pull request merged**